Sources/CoreMetrics/Metrics.swift

```diff
@@ -247,20 +253,85 @@ extension Recorder: CustomStringConvertible {
     }
 }
 
-/// A gauge is a metric that represents a single numerical value that can arbitrarily go up and down.
-/// Gauges are typically used for measured values like temperatures or current memory usage, but also "counts" that can go up and down, like the number of active threads.
-/// Gauges are modeled as `Recorder` with a sample size of 1 and that does not perform any aggregation.
-public class Gauge: Recorder {
+// MARK: - Gauge
+
+extension Gauge {
     /// Create a new `Gauge`.
     ///
     /// - parameters:
     ///     - label: The label for the `Gauge`.
     ///     - dimensions: The dimensions for the `Gauge`.
     public convenience init(label: String, dimensions: [(String, String)] = []) {
-        self.init(label: label, dimensions: dimensions, aggregate: false)
+        self.init(Recorder(label: label, dimensions: dimensions, aggregate: false))
+    }
+
+    /// Signal the underlying metrics library that this recorder will never be updated again.
+    /// In response the library MAY decide to eagerly release any resources held by this `Recorder`.
+    @inlinable
+    public func destroy() {
+        self.underlying.destroy()
+    }
+}
+
+/// A gauge is a metric that represents a single numerical value that can arbitrarily go up and down.
+/// Gauges are typically used for measured values like temperatures or current memory usage, but also "counts" that can go up and down, like the number of active threads.
+/// Gauges are modeled as `Recorder` with a sample size of 1 and that does not perform any aggregation.
+public final class Gauge: _SwiftMetricsSendable {
```

Comment:
Removing the inheritance makes this a semver major: is that intentional?

Reply:
I was hope we could get away without it since we keep the same public methods signature. any other ideas how to achieve this other than `@unchecked Sendable`?

Reply:
That's the only one I can think of.

Reply:
updated this